src/main/java/org/apache/solr/benchmarks/MetricsCollector.java

```diff
@@ -63,14 +57,23 @@ public void run() {
 						}
 					}
 					for (String path: metricsPaths) {
-						String group = path.split("/")[0];
+						String[] elements = path.split("/");
+						String group = elements[0];
 						JSONObject json = resp.get(group);
 
 						if (json != null) {
-							String key1 = path.split("/")[1];
-							String key2 = path.split("/")[2];
-							Double metric = json.getJSONObject("metrics").getJSONObject(key1).getDouble(key2);
-							metrics.get(node.getNodeName()).get(path).add(metric);
+							try {
+								json = json.getJSONObject("metrics");
+								int n = elements.length;
+								for (int i = 1; i < n - 1; ++i) {
+									json = json.getJSONObject(elements[i]);
+								}
+								Double metric = json.getDouble(elements[n - 1]);
+								metrics.get(node.getNodeName()).get(path).add(metric);
+							} catch (JSONException e) {
+								// some key, e.g., solr.core.fsloadtest.shard1.replica_n1 may not be available immediately
+								log.error("skipped metrics path {}:", path, e);
```

Comment:
I think we might want this to be:
log.error("skipped metrics path {}: {}", path, e);

Reply:
Should we pad these with dummy values when skipping so that we can co-relate the timings offsets later?

Reply:
> I think we might want this to be:
> log.error("skipped metrics path {}: {}", path, e);

later version of slf4j support this format to add exception with stack trace to the log. Using just "{}" will only output exception message. I updated to slf4j 1.7.30, and now it is logging properly.

Reply:
> Should we pad these with dummy values when skipping so that we can co-relate the timings offsets later?

Added dummy value.